report/report.go

```diff
@@ -111,3 +110,72 @@ func (r Report) LocalNetworks() []*net.IPNet {
 	}
 	return ipNets
 }
+
+// OriginTable produces a table (to be consumed directly by the UI) based on
+// an origin ID, which is (optimistically) a node ID in one of our topologies.
+func (r Report) OriginTable(originID string) (Table, bool) {
+	for nodeID, nodeMetadata := range r.Endpoint.NodeMetadatas {
```

Comment:
It stikes me this could be done like this:
```
if nodeMetadata, ok := r.Endpoint.NodeMetadatas[nodeID]; ok {
  return endpointOriginTable(nodeMetadata)
}
```